app/assets/javascripts/stats/status/statusService.js

```diff
@@ -0,0 +1,23 @@
+angular.module('ocWebGui.stats.status.service', ['ngResource'])
+  .factory('AgentStatusStats', function ($http) {
+    return {
+      stats: function (startDate, endDate, reportType) {
+        // swap dates other way around if startDate is after endDate
+        var startDateParts = startDate.split(".");
+        var startDateObject = new Date(startDateParts[2], startDateParts[1] - 1, startDateParts[0]);
+
+        var endDateParts = endDate.split(".");
+        var endDateObject = new Date(endDateParts[2], endDateParts[1] - 1, endDateParts[0]);
+
+        if (startDateObject > endDateObject) {
+          [startDate, endDate] = [endDate, startDate];
```

Comment:
Browsers or minifiers may not understand this. Safer to use a temporary variable.